backends/core/models.py

```diff
@@ -249,9 +316,48 @@ def destroy(self):
       Param.destroy(*param_ids)
     self.delete()
 
+  def get_status(self, max_retires=10):
+    key = str(self.pipeline_id) + '_' + str(self.id) + '_' + CACHE_KEY_STATUS
+    retries = 0
+    while retries < max_retires:
+      status = self.memcache_client.gets(key)
+      if status is None: 
+        '''The status is uninitialized in memcache, 
+          getting it from the database and
+          updating its value in memcache
+        '''
+        self.memcache_client.set(key, self.status, time=MEMCACHE_DEFAULT_EXPIRATION_TIME_SECONDS)
```

Comment:
I believe this set is subject to concurrency too. Can you make use of `_add_or_update_multi_to_cache()` instead?

Reply:
great observation. Since it's only one key to be added, I think the add method would be suitable here. done